core/lib/utils/mail/mail_address.dart

```diff
@@ -189,7 +222,7 @@ class MailAddress with EquatableMixin {
     } else {
       return MailAddress.fromParts(
           localPartWithoutDetails: getLocalPartWithoutDetails(),
-          localPartDetails: strictUrlEncode(localPartDetails),
+          localPartDetails: customUriEncode(localPartDetails),
```

Comment:
Are we sure this is 100% compliant with what is implemented on James server side?